### PR TITLE
pass i3s loader only required options

### PR DESCRIPTION
**Pull request by kirillovzah**
Issue https://github.com/visgl/loaders.gl/issues/1118

### CODE REVIEW

File: modules/i3s/src/lib/parsers/parse-i3s-tile-content.ts [hunk 1]
```diff
@@ -274,7 +286,7 @@ function normalizeAttribute(attribute: I3SMeshAttribute): I3SMeshAttribute {
   return attribute;
 }
 
-function parseHeaders(tileset: I3STilesetHeader, arrayBuffer: ArrayBuffer) {
+function parseHeaders(tileset: I3STilesetOptions, arrayBuffer: ArrayBuffer) {
```
Comment (ibgreen) [inline modules/i3s/src/lib/parsers/parse-i3s-tile-content.ts:289]: - If this is now an options object and not a tileset instance, can we call the parameter `options`
- The normal parameter order is `(arrayBuffer, options)`

File: modules/i3s/src/lib/parsers/parse-i3s-tile-content.ts [hunk 2]
```diff
@@ -394,7 +406,7 @@ function parseUint64Values(
   return new Uint32Array(values);
 }
 
-function parsePositions(attribute: I3SMeshAttribute, tile: I3STileHeader): Matrix4 {
+function parsePositions(attribute: I3SMeshAttribute, tile: I3STileOptions): Matrix4 {
```
Comment (ibgreen) [inline modules/i3s/src/lib/parsers/parse-i3s-tile-content.ts:409]: Rename to `options`?

**Review by belom88 — CHANGES_REQUESTED** [review]

File: modules/tiles/src/tileset/tile-3d.ts [hunk 3]
```diff
@@ -757,8 +757,18 @@ export default class TileHeader {
       case 'i3s':
         return {
           ...this.tileset.options.i3s,
-          tile: this.header,
-          tileset: this.tileset.tileset,
+          tile: {
+            content: this.header.content,
```
Comment (belom88) [inline modules/tiles/src/tileset/tile-3d.ts:761]: remove this too?

**Review by belom88 — CHANGES_REQUESTED** [review]

**Review by belom88 — APPROVED** [review]

**Review by ibgreen — COMMENTED** [review]
This changes return value of a lot of functions. But none of these functions have return value types. Please add such types, then I will be comfortable with merging this.

I will say that there was already some subtlety to how this was working as we are passing in a (possibly empty) tile object for these functions to populate and then return that object. Now it is returning something which can be different than the passed in object - so that original choice of passing in an empty object is getting even more confusing. 

Basically we are adding more complexity without addressing the initial complexity making it harder to maintain - but in some sense that is your problem now...

**Review by ibgreen — APPROVED** [review]
After discussion with @belom88 it was explained that `tile` is no longer the full tile but some cherrypicked props to enable worker processing.

My confusion comes from the variable `tile` not being renamed. Please address that in this or a separate PR and I will be OK with this.
